src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -66,21 +66,43 @@ void memory_snapshot_harness_generatort::validate_options(
       "--harness-type initialise-from-memory-snapshot");
   }
 
-  start_instruction = get_start_instruction(goto_model);
-  if(!start_instruction.has_value())
+  if(initial_source_location_line.empty() == initial_goto_location_line.empty())
   {
     throw invalid_command_line_argument_exceptiont(
-      "could not find the specified entry point", "--initial-location/line");
+      "choose either source or goto location to specify the entry point",
+      "--initial-source/goto-location");
+  }
+
```

Comment:
I think this needs a check that only one of goto location and source location is set?

Reply:
It's there somewhere (line 69).